app/code/core/Mage/Adminhtml/controllers/Sales/OrderController.php

```diff
@@ -277,13 +277,23 @@ public function addCommentAction()
                 $notify = isset($data['is_customer_notified']) ? $data['is_customer_notified'] : false;
                 $visible = isset($data['is_visible_on_front']) ? $data['is_visible_on_front'] : false;
 
-                $order->addStatusHistoryComment($data['comment'], $data['status'])
+                $comment = $data['comment'];
+
+                $session = Mage::getSingleton('admin/session');
+                $username = $session->getUser()->getUsername();
+                
+                if (!$notify && !$visible) {
+        	        $comment .= (!issempty($username)) ? " ({$username})" : ' (system)';
```

Comment:
Please fix `isempty`

Reply:
Changed to isset() myabe wiser than ($username)

Reply:
the variable is always set, even when false is returned. isn't empty() the right choice here?
https://www.virendrachandak.com/techtalk/php-isset-vs-empty-vs-is_null/